src/game_enemy.cpp

```diff
@@ -244,3 +247,65 @@ const lcf::rpg::EnemyAction* Game_Enemy::ChooseRandomAction() {
 	return nullptr;
 }
 
+bool Game_Enemy::IsSkillUsableForAI(int skill_id) const {
```

Comment:
RPG_RT does this by calling a function called `RPG_Battler::IsSkillEffectiveOn` which does a set of checks on each battler. The logic you have here is close but missing some edge cases. Also I need to do more testing to be sure.